src/sql-editor/standardSql/standardSQLCompletionItemProvider.ts

```diff
@@ -0,0 +1,39 @@
+import { Monaco } from '@grafana/ui';
+import { Operator, OperatorType, SQLCompletionItemProvider } from 'sql-editor/types';
+import { MACROS } from './macros';
+import { SQLMonarchLanguage } from './types';
+
+export function getStandardSQLCompletionProvider(
+  monaco: Monaco,
+  language: SQLMonarchLanguage
+): SQLCompletionItemProvider {
+  const provider: SQLCompletionItemProvider = { triggerCharacters: ['.', ' ', '$', ',', '(', "'"] };
+  if (language?.builtinFunctions.length) {
+    provider.supportedFunctions = () => language.builtinFunctions.map((f) => ({ id: f, name: f }));
+  }
+
+  const operators: Array<Operator> = [];
+  if (language?.comparisonOperators?.length) {
+    operators.push(
+      ...language.comparisonOperators.map((f) => ({
+        id: f.toLocaleLowerCase(),
+        operator: f,
+        type: OperatorType.Comparison,
+      }))
+    );
+  }
+
+  // some languages in the monaco language registry doesn't specify logical operators, only operators. if so, suggest them instead
```

Comment:
```suggestion
  // some languages in the monaco language registry don't specify logical operators, only operators. if so, suggest them instead
```

i see this answers my previous question i guess sorry! So when we say some languages, do we mean like some flavors of sql? Is that because of our own implementation? Would it make sense to have them all adhere to a specific meaning?

Reply:
yes is seems like all sql languages in the monaco registry define operators. logical operators/comparison operators is our own way of enhancing that to improve autocompletion experience